moma_mission/src/moma_mission/states/observation.py

```diff
@@ -66,3 +67,138 @@ def run(self):
 
         rospy.sleep(2.0)
         return 'Completed'
+
+class FOVSamplerState(StateRos):
+    """
+    Computes an observation point of view, that given a known object size (in terms of bounding box)
+    and known camera intrinsics, makes sure that the object is in the field of view and camera is pointig
+    at the object.
+    Assuming the camera Z is pointing to the object, the minimum conservative distance to keep the object in the field of 
+    view is s * focal_distance / image_width. We approximate the numerator with max(kx * f, ky * f) and the denominator with
+    min(width, height) as this information is readily available from the camera info topic. 
+    """
+    #TODO (giuseppe) this is computing the reference pose of the camera, but sending as reference pose for the tool
+    # -> must account for offset
+
+    def __init__(self, ns, outcomes=['Completed', 'Failure']):
+        StateRos.__init__(self, ns=ns, outcomes=outcomes)
+
+        # the name of the observation tf published
+        self.sample_name = self.get_scoped_param("sample_name", "observation")
+
+        # we use these names to resolve an available estimate in the current tf tree
+        self.map_frame = self.get_scoped_param("map_frame", "world")
+        self.object_frame = self.get_scoped_param("object_frame", "object")
+
+        # Use these frame ids to find the offset for the transform, which is where
+        # the controlled frame goes, not the camera frame.
+        self.camera_frame = self.get_scoped_param("camera_frame", "camera_optical_frame")
+        self.controlled_frame = self.get_scoped_param("controlled_frame", "controlled_frame")
+
+        # the approximate lumped size of the object bounding box
+        self.object_size = self.get_scoped_param("object_size", 1.0)
+
+        # how big in the image the object should look like. 1.0 means getting the closest to the object
+        self.image_fill_ratio = self.get_scoped_param("image_fill_ratio")
+
+        # needed for FOV computation
+        self.camera_info_topic = self.get_scoped_param("camera_info_topic", "/camera_info")
+        self.heading = self.get_scoped_param("heading", [1, 0, 0])
+        
+        # how much to look lateraly across all candiate observation poses (it is the phi angle in sperical coords)
+        self.lateral_view_angle = np.deg2rad(self.get_scoped_param("lateral_view_angle", 20))
```

Comment:
I think we should have a common definition of how we specify angle parameters throughout the MoMa stack. I have seen a mix of both, degrees and radians. For the spherical observer I defaulted to radians.

Reply:
For the state machine params I would rather use rad and use the `_deg` suffix. While in mpc those params are static, is very likely that here we want to repeadatily try values, and degrees are more difficult to mess-up and we have a more realistic intuition of what they mean.